### PR TITLE
Stairs are Dangerous

### PR DESCRIPTION
**Pull request by Geevies**
* Crates, closets, rollerbeds and wheelchairs now cannot safely traverse stairs.

### CODE REVIEW

**Comment by Geevies** [comment]
!review

File: code/game/objects/structures/crates_lockers/crates.dm [hunk 1]
```diff
@@ -85,8 +84,14 @@
...
-
-
+/obj/structure/closet/crate/stair_act()
+	if(!opened)
+		visible_message(SPAN_WARNING("\The [src] flies open as it bounces on the stairs!"))
+		for(var/thing in src)
+			var/atom/movable/AM = thing
+			AM.forceMove(get_turf(src))
+			AM.throw_at_random(TRUE, 1, 2)
+		open()
```
Comment (NonQueueingMatt) [inline code/game/objects/structures/crates_lockers/crates.dm:94]: wouldn't this happen for locked crates as well?

File: code/modules/multiz/structures.dm [hunk 2]
```diff
@@ -210,6 +210,10 @@
 				playsound(src, 'sound/effects/stairs_step.ogg', 50)
 				playsound(target, 'sound/effects/stairs_step.ogg', 50)
 
+/obj/structure/stairs/Crossed(obj/O)
+	if(istype(O))
+		O.stair_act()
```
Comment (NonQueueingMatt) [inline code/modules/multiz/structures.dm:215]: I think most of these should only happen if the dir of the object is heading towards the bottom of the stairs, tbh

**Comment by skull132** [comment]
As sue pointed out in discord, this may lead to troubles with jobs that need you to transport a load of shit:tm: to a small 2nd floor. Thoughts?

**Comment by Geevies** [comment]
> As sue pointed out in discord, this may lead to troubles with jobs that need you to transport a load of shit to a small 2nd floor. Thoughts?

For Investigations, adding a hovercrate, most likely. I already gave cargo a cargo hole and a hoist kit. Medical already has an elevator.

**Comment by Geevies** [comment]
This PR will return at a better time

**Comment by Geevies** [comment]
told you so

**Comment by DreamySkrell** [comment]
worth to note I think

ops has a lift spanning all decks
and there is a public access lift near security/hangar, also spanning all decks

**Comment by Sniblet** [comment]
I believe that hospital stretchers have been somewhat stair-friendly for about 600 years by the time of Aurora's setting. If you want to have some consequence for taking the stairs anyway, having stair_act hit when the holder is running sounds more reasonable and comes with some bonus interaction wherein pushing the stretcher is safe (does NOT make sense when going downstairs)

**Review by Arrow768 — APPROVED** [review]

**Comment by Greenjoe12345** [comment]
> I believe that hospital stretchers have been somewhat stair-friendly for about 600 years by the time of Aurora's setting. If you want to have some consequence for taking the stairs anyway, having stair_act hit when the holder is running sounds more reasonable and comes with some bonus interaction wherein pushing the stretcher is safe (does NOT make sense when going downstairs)

I had a thought of maybe giving first responders the hoverbeds, while the rest of the rollerbeds in medical are the normal ones

**Review by FluffyGhoster — APPROVED** [review]

**Comment by skull132** [comment]
> told you so

I got a notification about this PR being merged and was confused. Lmao, well played.

**Pull request merged**